BrainPortal/app/views/userfiles/_default_qc_panel.html.erb

```diff
@@ -23,14 +23,30 @@
 -%>
 
 <% if ! @userfile.is_locally_synced? %>
-  This file must be synced locally to view QC data.
+  <span class="warning">
+    This file must be synced locally to view QC data.
+  </span>
 <% else %>
-  <% if @userfile.is_a?(ImageFile) %>
-    <%= image_tag url_for(:action  => :content, :id => @userfile.id) %><br>
+  <% div_id_to_replace = "sub_viewer_#{@userfile.id}" %>
+  <% link = data_link(@userfile.name, @userfile, div_id_to_replace) %>
+  <% if link =~ /data-url=/%>
+    <div id="<%= div_id_to_replace %>">
+      <%= link %>
+    </div>
   <% else %>
     <span class="warning">
-    This file does not seem to contain any QC data, or no template is available to QC this file type.
+      This file does not seem to contain any QC data, or no template is available to QC this file type.
     </span>
   <% end %>
 <% end %>
 
+<script type="application/javascript" defer>
+  // Find the link inside the div and click it
+  <% if div_id_to_replace %>
```

Comment:
I feel uncomfortable about the scope of the variable `id_to_replace` though. It feels it is a local variable inside a code block well above in the HTML page, so maybe move its initialization out to be at the level of the entire page, or (better) move the entire script tag inside the `if link =~ /data-url/`

Reply:
My original comment before I suggested moving the script tag was that the condition `if div_id_to_replace` shoudl go AROUND the entire script tag, but that is moot if you move the script tag to the HTML above.